src/main/java/nl/tudelft/semgroup4/GameState.java

```diff
@@ -129,10 +125,6 @@ public void update(GameContainer container, StateBasedGame mainApp, int delta)
         }
 
         if (!currentGame.isPaused()) {
```

Comment:
Not in your diff, but since you change these update methods, and I notice it now:

You do an isPaused check here in the gamestate. So the game.update is only ever called if the game is not paused. There is no need to check the isPaused within the game.update before polling the inputs. (see SinglePlayerGame line 51 and MultiPlayerGame line 58).

< performance